### PR TITLE
[v7] Deprecate dependent attributes

### PR DESCRIPTION
**Pull request by doctrino**
## Description
These attributes should just be properties.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/transformations/__init__.py [hunk 1]
```diff
@@ -169,6 +174,14 @@ def __init__(
         self.tags = tags
         self._cognite_client = cast("CogniteClient", cognite_client)
 
+    @property
+    def has_source_oidc_credentials(self) -> bool:
+        return self.source_oidc_credentials is not None
+
+    @property
+    def has_destination_oidc_credentials(self) -> bool:
+        return self.destination_oidc_credentials is not None
```
Comment (haakonvt) [inline cognite/client/data_classes/transformations/__init__.py:183]: If they are set to `False`, these return `True`

Comment (doctrino) [reply cognite/client/data_classes/transformations/__init__.py:183]: What do you mean? The `self.destination_oidc_credentials` should bi set to `OidcCredentials`.

Comment (haakonvt) [reply cognite/client/data_classes/transformations/__init__.py:199]: I can't read 😂

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ methods are `json` and `yaml` serializable.
 - The `client.events.aggregate` use `client.events.aggregate_count` instead.
 - The `client.sequence.aggregate` use `client.sequence.aggregate_count` instead.
 - The `client.time_series.aggregate` use `client.time_series.aggregate_count` instead.
+- In `Transformations` attributes `has_source_oidc_credentials` and `has_destination_oidc_credentials`.
```
Comment (haakonvt) [inline CHANGELOG.md:42]: The sentence lacks an ending.

File: CHANGELOG.md [hunk 3]
```diff
@@ -77,6 +78,8 @@ with no easy way to add a prefix. Also, it no longer expands metadata by default
   `dict` but classes with attributes matching the API.
 - Calling `client.iam.token.inspect()` now gives an object `TokenInspection` with attribute `cababilities` of type `ProjectCapabilitiesList`
   instead of `list[dict]`
+- In data class `Transformation` the attribute `schedule`, `running_job`, and `last_running_job` `external_id` and `id`
+  are set to the `Transformation` `id` and `external_id` if not set. If they are set to a different value, a `ValueError` is raised.
```
Comment (haakonvt) [inline CHANGELOG.md:82]: A bit hard to read this one:
```suggestion
- In data class `Transformation` the attribute `schedule`, `running_job`, and `last_running_job`, `external_id` and `id`
  are set to the `Transformation` `id` and `external_id` if not set. If they are set to a different value, a `ValueError` is raised.
```

File: cognite/client/data_classes/transformations/__init__.py [hunk 4]
```diff
@@ -107,6 +107,7 @@ class Transformation(CogniteResource):
         source_session (SessionDetails | None): Details for the session used to read from the source project.
         destination_session (SessionDetails | None): Details for the session used to write to the destination project.
         tags (list[str] | None): No description.
+        **_ (Any): No description.
```
Comment (haakonvt) [inline cognite/client/data_classes/transformations/__init__.py:110]: Why do we need kwargs here?

File: cognite/client/data_classes/transformations/__init__.py [hunk 5]
```diff
@@ -148,11 +150,14 @@ def __init__(
         self.is_public = is_public
         self.ignore_null_fields = ignore_null_fields
         self.source_oidc_credentials = source_oidc_credentials
-        self.has_source_oidc_credentials = has_source_oidc_credentials or source_oidc_credentials is not None
+        if has_source_oidc_credentials or has_destination_oidc_credentials:
+            warnings.warn(
+                "The arguments 'has_source_oidc_credentials' and 'has_destination_oidc_credentials' are "
+                "deprecated and will be removed in a future version."
+                "These are now properties returning whether the transformation has source or destination oidc credentials set.",
+                DeprecationWarning,
```
Comment (haakonvt) [inline cognite/client/data_classes/transformations/__init__.py:158]: I vote for UserWarning, as 99/100 won't see the (correct) `DeprecationWarning`. Up to you

File: cognite/client/data_classes/transformations/__init__.py [hunk 6]
```diff
@@ -169,6 +174,30 @@ def __init__(
...
+        if self.last_finished_job:
+            self.last_finished_job.id = self.last_finished_job.id or self.id
+        if self.schedule and self.external_id != self.schedule.external_id:
+            raise ValueError("Transformation external_id must be the same as the schedule.external_id.")
+        if (
+            (self.schedule and self.id != self.schedule.id)
+            or (self.running_job and self.id != self.running_job.id)
+            or (self.last_finished_job and self.id != self.last_finished_job.id)
+        ):
+            raise ValueError("Transformation id must be the same as the schedule, running_job, last_running_job id.")
```
Comment (haakonvt) [inline cognite/client/data_classes/transformations/__init__.py:191]: Move to a `validate` method?

**Review by haakonvt — APPROVED** [review]
Some comments you may have a look at, then merge once you're happy 😉

**Pull request merged**